cloudinit/sources/DataSourceAzure.py

```diff
@@ -529,6 +543,9 @@ def _get_data(self):
         except Exception as e:
             LOG.warning("Failed to get system information: %s", e)
 
+        if self.distro is not None:
+            self.distro.networking.blacklist_drivers = BLACKLIST_DRIVERS
```

Comment:
I *think* this is OK, but wanted to check with @OddBloke about restoring a datasource from obj.pkl where distro.networking might not be present?  What did we do about that?

Further, if the distro.networking is there, would we need to check if it hasattr? before setting

Reply:
This is not really an issue for Azure datasource because the obj.pkl gets deleted on every boot (due to Azure's requirement to have network re-configured on boot).
I also did test on upgrading an existing VM with this new code and keeping the obj.pkl (by disabling the systemd unit that would delete the obj.pkl on boot) and it works fine. If the datasource was pickled the _get_data method would not get called since the data was cached

Reply:
From our IRC conversation: there might be an existing issue before my change with networking being an instance variable. This change, however, might make that bug easier to run into. @OddBloke mentioned he had a local commit that I can incorporate into this PR to address the issue altogether.